keyring/backends/Windows.py

```diff
@@ -100,7 +100,12 @@ def _set_password(self, target, username, password):
                           CredentialBlob=password,
                           Comment="Stored using python-keyring",
                           Persist=win32cred.CRED_PERSIST_ENTERPRISE)
-        win32cred.CredWrite(credential, 0)
+        try:
+            win32cred.CredWrite(credential, 0)
+        except pywintypes.error as e:
+            import sys
+            type, exc, tb = sys.exc_info()
+            raise PasswordSetError,("Max password length exceeded",type,exc),tb
```

Comment:
Probably this needs to use [six.reraise](https://pythonhosted.org/six/#six.reraise) for Python 3 compatibility.

Reply:
Does keyring already have six as a dependency?  Or are you saying I should add it as one?

Reply:
Oh, interesting. I was assuming six was a dependency, but I see now it's not. I see that the py27compat module has some code borrowed from six.py, so probably best to repeat that pattern and copy [these lines](https://github.com/benjaminp/six/blob/6845a1b7670aeea9f5b19c3bf84aa6419f047a77/six.py#L684-L716) into py27compat.py.